### PR TITLE
Promisify function properties as well

### LINKED ISSUE #2: module.exports is a function?
```js
function foo(callback) {}
foo.bar = function bar(callback) {};
module.exports = foo;
```

How does promisify-all work with this module?

### PR DESCRIPTION
**Pull request by SimenB**
Fixes #2

/cc @ljharb

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,4 +13,18 @@ module.exports = obj => {
...
     .reduce((acc, curr) => Object.assign(acc, curr), {});
+}
+
+module.exports = obj => {
+  const promisifiedObject = promisifyObject(obj);
+
+  if (typeof obj === 'function') {
+    const promisiedFunction = promisify(obj);
+
+    Object.assign(promisiedFunction, promisifiedObject);
```
Comment (SimenB) [inline index.js]: This makes me feel dirty, how illegal is this? I can do `Object.entries` again, I suppose - is that better?

Comment (ljharb) [inline index.js]: This seems fine to me, but it's probably be better to check the type and only do `promisifyObject` on `obj` in the non-function case. (`s/promisied/promisified/g` tho)

Comment (SimenB) [inline index.js]: Then `foo.bar` won't be promisified, though. Wasn't that the point?

Comment (SimenB) [inline index.js]: Typo fixed, but that made this comment "outdated" 😢

**Comment by SimenB** [comment]
Thanks for the feedback and review! ❤️

>be aware that modules can export primitives

#4

>as well as constructable functions

That bit me (with a `class`, but of course it'd hit other functions as well). Any ideas on how to handle that?

**Comment by ljharb** [comment]
Since constructors can't be async, I'm not really sure.

**Pull request merged**